23937 change button text continuation auth

TVWerdal opened the pull request:
Issue #:  bcgov/entity#23937 

_Description of changes:_

- Updated wording for action button as a continue in moves through draft stages
- Updated wording of "Status" in table to match requirements here: https://www.figma.com/design/rNe2jn8wyFykKRzwiBo6ar/Continuation-In-V6?node-id=100-4595&p=f&t=afbXH10lhe60tOCM-0
 
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

TVWerdal commented:
/gcbrun

bcregistry-sre commented:
Temporary Url for review: https://business-registry-dev--pr-61-pr4g2pat.web.app

JazzarKarim commented:
Please Thayne if you can rebase and re-update that package.json version 🙏

JazzarKarim commented:
So, I'm looking at examples in the table:
![image](https://github.com/user-attachments/assets/cc68d144-0ccd-44b6-bb24-4756b720e6f5)

And the figma link, I think things looks great 👍

JazzarKarim commented on business-registry-dashboard/app/components/Table/AffiliatedEntity/Action.vue:
```diff
@@ -154,8 +154,29 @@ function getPrimaryActionLabel (item: Business): string {
+      // For now seperating out Cont In's, but leaving in ability to switch messages to other filing types
+      if (item.corpType.code === CorpTypes.CONTINUATION_IN) {
+        switch (true) {
+          case (item.draftStatus === 'AWAITING_REVIEW'):
```
Can we use an enum for these values?

TVWerdal replied:
Sure, will implement that

JazzarKarim replied:
Thanks! Resolving.

ArwenQin commented:
It looks good and I agree with using an enum for these values

severinbeauvais reviewed (changes requested):
There is an existing enum, in our shared components repo, that you should use instead.

severinbeauvais commented on business-registry-dashboard/app/enums/business-state.ts:
We already have a shared enum for this, but it's missing a lots of items. Would you mind updating the shared enum and importing it in this UI instead?

https://github.com/bcgov/bcrs-shared-components/blob/main/src/enums/entity-states.ts

severinbeauvais replied:
Wait, I'm confusing this with the next enum.

I believe THIS enum does not exist in the shared repo, but it would be nice to have there.

The next enum does exist although it's named EntityStates. It would be nice to have it complete in the shared repo (per my comment above).

I suggest putting both enums in the same file in the shared repo since they really go together.

Don't rename the shared enum as other UIs probably use it.

TVWerdal replied:
Pull request in place for enum update in shared repo: https://github.com/bcgov/bcrs-shared-components/pull/266

TVWerdal replied:
Switched this bit of code to use shared components.

TVWerdal commented on business-registry-dashboard/app/utils/affiliations.ts:
```diff
@@ -98,8 +98,29 @@ export const affiliationType = (business: Business): string => {
+      }
+      // For now seperating out Cont In's, but leaving in ability to switch messages to other filing types
+      if (business.corpType.code === CorpTypes.CONTINUATION_IN) {
+        switch (true) {
```
Updated in most recent version

severinbeauvais commented on business-registry-dashboard/app/components/Table/AffiliatedEntity/Action.vue:
```diff
@@ -154,8 +154,27 @@ function getPrimaryActionLabel (item: Business): string {
+          case (EntityStates.APPROVED):
+            return t('labels.resumeApplication')
+          case (EntityStates.AWAITING_REVIEW):
+            return t('labels.openApplication')
```
If it's awaiting review then it's still an authorization.

Should this say "open authorization"?

TVWerdal replied:
Meant to include this in my comment and forgot - this comes out of a chat with Jacqueline:

•	Couple of changes, we are using the term ‘Application’ instead of authorization because under the type column on the BRD we are calling this a Continuation Application and treating the authorization as the first step in a single process - the continuation application process. If we want to change the wording to ‘open authorization’, we will probably need to change some of the wording under the type column and that was outside the scope when the design as initially discussed. 
•	Let’s show  “Manage Business” for the rejected state. The designs were inconsistent – this was only showing in the one section on figma – sorry for the confusion 
•	For the pending and paid are status’ could we just put Manage Business? Since the user is just waiting for the filer to process, and I don’t think these status’ will show on the BRD UI (under the status column on the table)?

severinbeauvais replied:
First bullet: gotcha. Leave it as is.

Second bullet: but it's not a business yet. Ask Jacqueline if this is OK.

Third bullet: it's still not a business. What status does show in the BRD when the filing is pending (awaiting payment) or paid (awaiting Filer)?

severinbeauvais replied:
> Third bullet: it's still not a business. What status does show in the BRD when the filing is pending (awaiting payment) or paid (awaiting Filer)?

What do we say for other bootstrap filings that are pending or completed?

TVWerdal replied:
Jacqueline has provided us with the following, which I worked from for the most recent commit : 

STATUS | Status as it displays on BRD | Description | Button Text | Secondary Action(s)
-- | -- | -- | -- | --
DRAFT | Draft | Authorization started but not completed. | Resume Draft | Delete Application
AWAITING REVIEW | Pending Staff Review | Authorization Pending Approval | Open Application | Remove from Table
CHANGE REQUESTED | Change Requested (with icon) | Changes requested to authorization | Make Changes | Remove from Table
REJECTED | Rejected | Authorization Rejected | Open Application | Remove from Table
APPROVED | Approved | Authorization Approved | Resume Application | Remove from Table
DRAFT | Draft | Authorization Approved and draft Continuation Application   started | Resume Draft | Remove from Table
PENDING | Active | Continuation Submitted and awaiting payment | Manage Business | Remove from Table
PAID | Active | Continuation Submitted and awaiting Filer | Manage Business | Remove from Table
COMPLETED | Continuation Submitted and Paid | Continuation Submitted, Filed and Paid | Manage Business | Remove from Table

severinbeauvais replied:
I'd resolve this conversation but it contains useful review material 😀

severinbeauvais commented on business-registry-dashboard/app/components/Table/AffiliatedEntity/Action.vue:
```diff
@@ -154,8 +154,27 @@ function getPrimaryActionLabel (item: Business): string {
+          case (EntityStates.REJECTED):
+            return t('labels.manageBusiness')
+          default:
+            return t('labels.resumeDraft')
```
I'd prefer to see specific states use the "resume draft" text, and maybe "manage business" can be the default (no need for pending/paid/rejected?).

TVWerdal replied:
Changed in most recent commit

severinbeauvais commented on business-registry-dashboard/app/utils/affiliations.ts:
```diff
@@ -98,10 +102,31 @@ export const affiliationType = (business: Business): string => {
 /** Returns the status of the affiliation. */ // TODO: add i18n for these states
 export const affiliationStatus = (business: Business): string => {
```
Please update the return type of this function.

severinbeauvais replied:
Never mind. EntityStateStatus is actually a string. Resolving.

severinbeauvais commented on business-registry-dashboard/app/enums/business-state.ts:
How about deleting this duplicate enum and using the imported, shared one instead?

TVWerdal replied:
This is done in latest commit

JazzarKarim commented:
Please Thayne if you can rebase again and update that package.json as there's a conflict now.

severinbeauvais commented:
> ![image](https://private-user-images.githubusercontent.com/122301442/407939399-cc68d144-0ccd-44b6-bb24-4756b720e6f5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzg2MjM1ODMsIm5iZiI6MTczODYyMzI4MywicGF0aCI6Ii8xMjIzMDE0NDIvNDA3OTM5Mzk5LWNjNjhkMTQ0LTBjY2QtNDRiNi1iYjI0LTQ3NTZiNzIwZTZmNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIwM1QyMjU0NDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zYWFiOWI5ZTU0OThmODViYjdhNDVlYTE2NmRiZWVkNTRiODU2ZTA0N2ZiMjYxOTcyYjlhOTQwNmNjYTdlYjJiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.YELJnoBJKDxIiFUp3yEumkNrYrka1dQ8UnMboTCE5Mk)

What do the drop-down actions say for authorization vs application? Is that in scope for this ticket?

severinbeauvais commented:
Hey, Karim, can you get an example screenshot again?

JazzarKarim commented:
/gcbrun

JazzarKarim commented:
> Hey, Karim, can you get an example screenshot again?

The cloud build run is failing. I think that might be because the branch has conflicts with main. As soon as that's fixed, I'll do that!

TVWerdal commented:
> > ![image](https://private-user-images.githubusercontent.com/122301442/407939399-cc68d144-0ccd-44b6-bb24-4756b720e6f5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzg2MjM1ODMsIm5iZiI6MTczODYyMzI4MywicGF0aCI6Ii8xMjIzMDE0NDIvNDA3OTM5Mzk5LWNjNjhkMTQ0LTBjY2QtNDRiNi1iYjI0LTQ3NTZiNzIwZTZmNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIwM1QyMjU0NDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zYWFiOWI5ZTU0OThmODViYjdhNDVlYTE2NmRiZWVkNTRiODU2ZTA0N2ZiMjYxOTcyYjlhOTQwNmNjYTdlYjJiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.YELJnoBJKDxIiFUp3yEumkNrYrka1dQ8UnMboTCE5Mk)
> 
> What do the drop-down actions say for authorization vs application? Is that in scope for this ticket?

Currently every record says "Delete Continuation Application" regardless of the current state of the authorization/application. This will be addressed as part of https://github.com/bcgov/entity/issues/24079 and related tickets.

![image](https://github.com/user-attachments/assets/f4579132-4bfc-42cf-8952-4270c7a11a57)

severinbeauvais commented:
> > What do the drop-down actions say for authorization vs application? Is that in scope for this ticket?
> 
> Currently every record says "Delete Continuation Application" regardless of the current state of the authorization/application. This will be addressed as part of [bcgov/entity#24079](https://github.com/bcgov/entity/issues/24079) and related tickets.
> 
> ![image](https://private-user-images.githubusercontent.com/58612694/410116261-f4579132-4bfc-42cf-8952-4270c7a11a57.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzg3ODEzNDgsIm5iZiI6MTczODc4MTA0OCwicGF0aCI6Ii81ODYxMjY5NC80MTAxMTYyNjEtZjQ1NzkxMzItNGJmYy00MmNmLTg5NTItNDI3MGM3YTExYTU3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjA1VDE4NDQwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU3Yjc5Mzc2NmI4YzhiYjNmNTg2ZGI5NzFlOWRiMmVjYWIwNzU2ZmE5Mjk1ZTE4NWM4M2YyMDQzYjI1MWY5OTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.qS7j_686ymyMoM7CiFf9BmxnBZ4Sei9sCUz1Orkvjrc)

OK by me. Please add a comment to 24079 to keep track of the work needed. I know there are a few other tickets that might be "closed by the resolution of 24102" and I don't want 24079 to be closed if there's work remaining.

severinbeauvais commented:
Thayne, this PR's title and description refer to 3 different tickets. (I updated the issue number syntax in the description.) Please resolve :)

TVWerdal commented:
> Thayne, this PR's title and description refer to 3 different tickets. (I updated the issue number syntax in the description.) Please resolve :)

I looked at this -  bcgov/entity#23935 was split out into two separate tickets, one for ui (bcgov/entity#23937) and one for lear. Essentially the extension of that ticket. bcgov/entity#24102 was the previous ticket I had looked at. But it was for the dashboard UI not the BRD. I guess I got that mixed up somewhere in the naming scheme here. It is unrelated to this change. Updated the description to note that these changes are for bcgov/entity#23937

JazzarKarim commented:
/gcbrun

bcregistry-sre commented:
Temporary Url for review: https://business-registry-dev--pr-61-pr4g2pat.web.app

JazzarKarim commented:
> Hey, Karim, can you get an example screenshot again?

![image](https://github.com/user-attachments/assets/98105676-8a9c-43f0-b0ca-76e7c6afb59c)

So when approved, it says "Resume Application".
When pending staff review, it says "Open Application".
When draft, it's just resume draft. 

Should be good.

JazzarKarim reviewed (approved):
LGTM. Please wait for another approval before merging!

severinbeauvais commented:
> Updated the description to note that these changes are for 23937

What about the title?

severinbeauvais commented on business-registry-dashboard/package.json:
```diff
@@ -49,6 +49,7 @@
   },
   "dependencies": {
     "@bcrs-shared-components/interfaces": "^1.1.16",
+    "@bcrs-shared-components/enums": "^1.1.20",
```
Remove `^` from this dependency and the one above -- we don't want npm to automatically update these without our knowledge.

Or, maybe interfaces and enums is OK. Karim, what do you think?

JazzarKarim replied:
I think it's OK for enums and interfaces as they're not UI components themselves but honestly I think it would be better to remove the `^` just to be safe. Maybe down the line we update one of the enums here, change it's values, but forget to fix here.

TVWerdal replied:
I agree it could cause issues, I may as well remove these ^ while I update the version, push it all in one more commit.

severinbeauvais commented on business-registry-dashboard/package.json:
Please update the app version again.

severinbeauvais reviewed (approved):
LGTM
